src/map/script.cpp

```diff
@@ -16769,21 +16775,16 @@ BUILDIN_FUNC(equip) {
 	if (!script_charid2sd(3,sd))
 		return SCRIPT_CMD_FAILURE;
 
-	t_itemid nameid = script_getnum(st,2);
-	std::shared_ptr<item_data> id = item_db.find(nameid);
-
-	if (id != nullptr) {
-		int i;
-
-		ARR_FIND( 0, MAX_INVENTORY, i, sd->inventory.u.items_inventory[i].nameid == nameid );
-		if (i < MAX_INVENTORY) {
-			pc_equipitem(sd,i,id->equip);
-			script_pushint(st,1);
-			return SCRIPT_CMD_SUCCESS;
-		}
+	uint64 id = script_getnum64(st,2);
+	int i;
+	ARR_FIND( 0, MAX_INVENTORY, i, sd->inventory.u.items_inventory[i].nameid == (t_itemid)id || sd->inventory.u.items_inventory[i].unique_id == id);
```

Comment:
I wouldn't overload `equip`, instead make a new command called `equip_uniqueid`, then only search for `unique_id`s there.